components/dashboard/src/service/service.tsx

```diff
@@ -164,8 +165,9 @@ export class IDEFrontendService implements IDEFrontendDashboardService.IServer {
     private ownerId: string | undefined;
     private user: User | undefined;
     private ideCredentials!: string;
+    private workspace!: Workspace;
```

Comment:
I'm wondering a bit why this needs to be an instance member, as it _seems_ it's only used [here](https://github.com/gitpod-io/gitpod/pull/19213/files#diff-c8930e07c7cd1e6e259fdef732fcb303539923abb895879cd31664a5379d2d51R224) (to have one initial `Workspace` shape to fallback to. Could it be passed per context?

Or is this just the pattern we are doing here? (Not familiar with this code).

Reply:
We use `this.workspace` inside `parseInfo` function too. Some metadata fields of workspace are required